src/cli.rs

```diff
@@ -96,6 +103,19 @@ impl TrinConfig {
         } else {
             None
         };
+        let private_key = if matches.is_present("private_key") {
+            let hex_private_key = value_t!(matches.value_of("private_key"), String)?;
+            match hex_private_key.len() {
+                64 => (),
+                val => panic!(
```

Comment:
Maybe could return error instead of panic?

Reply:
From what I [read](https://doc.rust-lang.org/book/ch09-00-error-handling.html) - `panic!` is for unrecoverable errors and `Err` is for recoverable errors. My assumption was that if a user enters an invalid key - we want to terminate and report. Similarly, we panic with other invalid cli args. 

If I understand your suggestion correctly, you're suggesting that we want to use a recoverable error? So that trin will continue to run with an invalid key? Seems like this could be problematic, since trin'll generate a random key if the user-provided key is invalid, and then the node has a different identity from what the user is expecting.

In terms of ux, imo it seems like the `panic!` output is slightly less muddled than `Err`

`panic!`
```
thread 'main' panicked at 'Invalid private key length: 6, expected 32 byte hexstring', src/cli.rs:110:24
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

`Err`
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Invalid private key length: 6, expected 32 byte hexstring"', src/cli.rs:121:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Merging for now, but i'll circle back to this in a follow up pr if we want to improve the cli-arg validation strategy